### PR TITLE
Added template.json

### PR DESCRIPTION
**Pull request by adam230594**
Added ARM template - app service, app service certificate, app service plan and application insights

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                        "value": "[variables('appServicePlanName')]"
+                    },
+                    "aspLocation": {
+                        "value": "West Europe"
+                    },
+                    "aspSize": {
+                        "value": "1"
+                    },
+                    "aspInstances": {
+                        "value": 1
```
Comment (chelnak) [inline azure/template.json]: Should be a parameter as 2 instances is best practice for prod.

File: azure/template.json [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                },
+                "parameters": {
+                    "appServicePlanName": {
+                        "value": "[variables('appServicePlanName')]"
+                    },
+                    "aspLocation": {
+                        "value": "West Europe"
+                    },
+                    "aspSize": {
+                        "value": "1"
```
Comment (chelnak) [inline azure/template.json]: Should be a parameter as the app might need to be scaled to different tiers in different environments.

File: azure/template.json [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                "templateLink": {
+                    "uri": "[concat(variables('deploymentUrlBase'), 'app-service-plan.json')]",
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "appServicePlanName": {
+                        "value": "[variables('appServicePlanName')]"
+                    },
+                    "aspLocation": {
+                        "value": "West Europe"
```
Comment (chelnak) [inline azure/template.json]: [resourceGroup().location] here?

File: azure/template.json [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+                    "aspInstances": {
+                        "value": 1
+                    }
+                }
+            }
+        },
+        {
+            "apiVersion": "2017-05-10",
+            "name": "app-service-certificate",
+            "type": "Microsoft.Resources/deployments",
```
Comment (chelnak) [inline azure/template.json]: Should have a condition and only deploy if a certificate name is passed

"condition": "[greater(length(parameters('certificateName')), 0)]",

File: azure/template.json [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+            "type": "string"
+        },
+        "appServicePlanResourceGroup": {
+            "type": "string"
+        },
+        "customHostName": {
+            "type": "string"
+        },
+        "certificateName": {
+            "type": "string"
```
Comment (chelnak) [inline azure/template.json]: should have a default value property that is set to empty

defaultValue: ""

File: azure/template.json [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+                                "name": "APPINSIGHTS_INSTRUMENTATIONKEY",
+                                "value": "[reference('app-insights').outputs.InstrumentationKey.value]"
+                            }
+                        ]
+                    },
+                    "customHostName": {
+                        "value": "[parameters('customHostName')]"
+                    },
+                    "certificateThumbprint": {
+                        "value": "[reference('app-service-certificate').outputs.certificateThumbprint.value]"
```
Comment (chelnak) [inline azure/template.json]: Should only deploy if the certificate deployment has happened:

"value": "[if(greater(length(parameters('customHostname')), 0), reference('app-service-certificate', '2018-11-01').outputs.certificateThumbprint.value, '')]"

File: azure/template.json [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "serviceName": {
+            "type": "string"
+        },
+        "appServicePlanResourceGroup": {
+            "type": "string"
+        },
+        "customHostName": {
+            "type": "string"
```
Comment (chelnak) [inline azure/template.json]: should have a default value property that is set to empty

defaultValue: ""

File: azure/template.json [hunk 8]
```diff
@@ -0,0 +1,183 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "serviceName": {
+            "type": "string"
+        },
+        "appServicePlanResourceGroup": {
+            "type": "string"
```
Comment (chelnak) [inline azure/template.json]: should have a default value property that is set to empty

defaultValue: ""

or is this parameter event needed?

File: azure/template.json [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+        },
+        "customHostName": {
+            "type": "string"
+        },
+        "certificateName": {
+            "type": "string"
+        },
+        "keyVaultName": {
```
Comment (chelnak) [inline azure/template.json]: should have a default value property that is set to empty

defaultValue: ""

File: azure/template.json [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+            "type": "string"
+        },
+        "keyVaultResourceGroup": {
```
Comment (chelnak) [inline azure/template.json]: should have a default value property that is set to empty

defaultValue: ""

File: azure/template.json (hunk 2 above)
Comment (chelnak) [reply azure/template.json]: New parameter should have a default value which can be set to 1

defaultValue: "1"

File: azure/template.json (hunk 1 above)
Comment (chelnak) [reply azure/template.json]: New parameter should have a default value which can be set to 1

defaultValue: 1

File: azure/template.json [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "appServiceName": {
+                        "value": "[variables('appServiceName')]"
+                    },
+                    "appServicePlanName": {
+                        "value": "[variables('appServicePlanName')]"
+                    },
+                    "appServicePlanResourceGroup": {
```
Comment (chelnak) [inline azure/template.json]: See comment on parameter

File: azure/template.json (hunk 7 above)
Comment (adam230594) [reply azure/template.json]: Don't think this is needed as the master template already has a default value for it

https://github.com/SkillsFundingAgency/das-platform-building-blocks/blob/master/templates/app-service.json#L33

File: azure/template.json (hunk 8 above)
Comment (adam230594) [reply azure/template.json]: replaced with [resourceGroup().name]

File: azure/template.json (hunk 6 above)
Comment (adam230594) [reply azure/template.json]: Do you mean if(greater(length(parameters('certificateName')), 0) ?

Seems like if the customHostName parameter is missing the master template will not use the certificateThumbprint value anyway due to this condition

https://github.com/SkillsFundingAgency/das-platform-building-blocks/blob/master/templates/app-service.json#L156

Comment (chelnak) [reply azure/template.json]: happy with that 👍

**Pull request merged**